libnmstate/nm/nmclient.py

```diff
@@ -77,7 +85,26 @@ def client(refresh=False):
             raise error.NmstateDependencyError(
                 "Missing introspection data for libnm"
             )
-    return _nmclient
+    return _nmclient.client
+
+
+def nmclient_context(func):
+    """
+    Decorator for handling NM.Client creation and clean up
+    """
+
+    @functools.wraps(func)
+    def wrapped(*args, **kwargs):
+        try:
+            _delete_client()
+            # Delete NM.Client is required, if not, previous changed interfaces
+            # might not updated in current NM.Client caches.
```

Comment:
I expected here the creation of the client. This seems to cause it implicitly at the usage.
I think it is dangerous and would have preferred allowing to create here the client if it does not exists yet and in other places just to consume it.

If we do not treat this now, we better track it and prioritize as it may hurt us later.

Reply:
This line is required to forcely refresh on changed IP stack.

For now, we don't have problem, but https://nmstate.atlassian.net/browse/NMSTATE-308 is created to track the following fix up.